### PR TITLE
[#176924316] Add EYCA activation API

### PR DESCRIPTION
**Pull request by AleDore**
#### List of Changes
- Add `startEycaActivation` Api specs
- Add `getEycaActivation` Api specs
- Add tests

#### Motivation and Context
While a citizen requests for a CGN and he's between 18 up to 30 years old, he wants also activate an EYCA card ( CGN and EYCA are co-badged ), but in case of something goes wrong in EYCA activation, it must be possible to call directly EYCA activation from APP ( user retries ), see this [PR](https://github.com/pagopa/io-functions-cgn/pull/20) for function's side implementation.

#### How Has This Been Tested?
It has been tested by performing unit tests and integration test.

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: src/services/__tests__/cgnService.test.ts [hunk 1]
```diff
@@ -473,3 +490,222 @@ describe("CgnService#getCgnActivation", () => {
...
+
+    expect(res).toMatchObject({
+      kind: "IResponseSuccessRedirectToResource"
+    });
+  });
+
+  it("should handle a success Accepted response", async () => {
+    mockStartEycaActivation.mockImplementationOnce(() =>
+    t.success({status: 202})
+  );
```
Comment (BurnedMarshal) [inline src/services/__tests__/cgnService.test.ts:619]:
```suggestion
      t.success({status: 202})
    );
```

File: src/services/cgnService.ts [hunk 2]
```diff
@@ -183,4 +184,85 @@ export default class CgnService {
...
+          case 500:
+            return ResponseErrorInternal(readableProblem(response.value));
+          default:
+            return ResponseErrorStatusNotDefinedInSpec(response);
+        }
+      });
+    });
+
+  /**
+   * Get EYCA's activation status for the logged user.
```
Comment (BurnedMarshal) [inline src/services/cgnService.ts:235]:
```suggestion
   * Get EYCA's activation status detail for the logged user.
```

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]
Some small changes

File: api_cgn.yaml [hunk 3]
```diff
@@ -85,6 +85,57 @@ paths:
...
+          description: Service unavailable.
+          schema:
+            $ref: "#/definitions/ProblemJson"
+    get:
+      operationId: getEycaActivation
+      summary: |
+        Get EYCA activation process' status
+      description: |
+        Get informations about an EYCA activation process
+      responses:
```
[the remark this answers is not in the document]
Comment (AleDore) [reply api_cgn.yaml:130]: Yep you're right

**Review by BurnedMarshal — APPROVED** [review]
lgtm

**Pull request merged**